helpers/middleman/blog.rb

```diff
@@ -4,25 +4,30 @@
 module MiddlemanBlogHelpers
 
   def blog_posts(order: nil)
-    case order
-    when :youngest_first
-      data.blog.posts.values.sort do |a,b|
-        a['created_at'] <=> b['created_at']
+    @blog_posts ||= {}
+    return @blog_posts[order] if @blog_posts[order]
+
+    @blog_posts[order] ||= case order
+      when :youngest_first
+        data.blog.posts.values.sort do |a,b|
+          a['created_at'] <=> b['created_at']
+        end
+      when :oldest_first
+        data.blog.posts.values.sort do |a,b|
+          b['created_at'] <=> a['created_at']
+        end
+      else
+        data.blog.posts
       end
-    when :oldest_first
-      data.blog.posts.values.sort do |a,b|
-        b['created_at'] <=> a['created_at']
-      end
-    else
-      data.blog.posts
-    end
   end
 
   def post_url(post)
     return unlocalized_url("blog/#{BlogHelper::url_slug(post)}")
   end
 
   def blog_categories
+    return @blog_categories if @blog_categories
```

Comment:
This is so...

The Ruby way would be to

```ruby
def blog_categories
  @blog_categories ||= begin
    ..
  end
end
```

Reply:
Ah, alright.  the `begin; end` was what I missed – didn't want to run the other crap.